python/packages/autogen-ext/src/autogen_ext/tools/web_search/_duckduckgo_search.py

```diff
@@ -0,0 +1,280 @@
+# DuckDuckGo Search Tool created by Varad Srivastava
+from typing import Dict, List, Optional, Union
+from urllib.parse import parse_qs, quote_plus, unquote, urljoin, urlparse
+
+import html2text
+import httpx
+from autogen_core import CancellationToken
+from autogen_core.tools import BaseTool
+from bs4 import BeautifulSoup, NavigableString, Tag
+from pydantic import BaseModel, Field
+
+
+class DuckDuckGoSearchArgs(BaseModel):
+    """Arguments for DuckDuckGo search."""
+
+    query: str = Field(..., description="Search query string")
+    num_results: int = Field(default=3, description="Number of results to return (max 10)")
+    include_snippets: bool = Field(default=True, description="Include result snippets in output")
+    include_content: bool = Field(default=True, description="Include full webpage content in markdown format")
+    content_max_length: Optional[int] = Field(
+        default=10000, description="Maximum length of webpage content (if included)"
+    )
+    language: str = Field(default="en", description="Language code for search results (e.g., en, es, fr)")
+    region: Optional[str] = Field(default=None, description="Optional region code for search results (e.g., us, uk)")
+    safe_search: bool = Field(default=True, description="Enable safe search filtering")
+
+
+class DuckDuckGoSearchResult(BaseModel):
+    """Result from DuckDuckGo search."""
+
+    results: List[Dict[str, str]] = Field(description="List of search results")
+
+
+class DuckDuckGoSearchTool(BaseTool[DuckDuckGoSearchArgs, DuckDuckGoSearchResult]):
+    """
+    A tool for performing DuckDuckGo web searches.
+
+    This tool uses DuckDuckGo's HTML interface to perform web searches without requiring
+    an API key. It can optionally fetch and convert webpage content to markdown format.
+
+    Example:
+        ```python
```

Comment:
Can you use RST code blocks? See examples from other API doc strings. 

The doc string must pass Pyright lint check. You can create a script in a separate file first, then, use pyright to check the script. Once there is no error, copy and paste the script back to the API doc.